### PR TITLE
Add S.C.I. and S.R.M. dependencies

### PR DESCRIPTION
**Pull request by ellahathaway**
Fixes https://github.com/dotnet/source-build/issues/3968 by adding System.Collections.Immutable and System.Reflection.Metadata to Version.Details and updating the versions to 8.0.

[Build with changes](https://dev.azure.com/dnceng/internal/_build/results?buildId=2354871&view=logs&j=4cede3bc-29a7-5d91-52ef-402f98629c03&t=dd97ae34-3027-5020-e259-ff298adf2f8e) (internal link)

### CODE REVIEW

File: eng/Versions.props [hunk 1]
```diff
@@ -63,11 +63,11 @@
     <MicrosoftiOSTemplatesVersion>15.2.302-preview.14.122</MicrosoftiOSTemplatesVersion>
     <MicrosoftiOSTemplatesVersion160527>16.0.527</MicrosoftiOSTemplatesVersion160527>
     <!-- Keep this version in sync with what msbuild / VS ships with. -->
-    <SystemCollectionsImmutableVersion>7.0.0</SystemCollectionsImmutableVersion>
+    <SystemCollectionsImmutableVersion>8.0.0</SystemCollectionsImmutableVersion>
```
Comment (mthalman) [inline eng/Versions.props:66]: Unless something has changed after https://github.com/dotnet/arcade/pull/14115, I think these need to remain at 7.0.0. Is that correct, @ViktorHofer?

Comment (ViktorHofer) [reply eng/Versions.props:66]: While VS components (roslyn & msbuild) just recently upgraded to SRM&SCI 8.0.0, the VS installation used i.e. when signing doesn't yet have those changes. So this needs to stay at 7.0.0 for a few more months.

Comment (mthalman) [reply eng/Versions.props:66]: Yeah, so the changes in Versions.props can be reverted and then the values in Version.Details.xml should be for 7.0.0.

**Review by ViktorHofer — COMMENTED** [review]
Q: Will this fix the errors in this PR? https://github.com/dotnet/runtime/pull/97087

>  /__w/1/s/artifacts/sb/package-cache/microsoft.dotnet.genfacades/9.0.0-beta.24066.3/build/Microsoft.DotNet.GenFacades.NotSupported.targets(56,5): error MSB4018: The "NotSupportedAssemblyGenerator" task failed unexpectedly. [/__w/1/s/artifacts/sb/src/src/libraries/System.IO.Pipes.AccessControl/src/System.IO.Pipes.AccessControl.csproj::TargetFramework=net9.0]
  /__w/1/s/artifacts/sb/package-cache/microsoft.dotnet.genfacades/9.0.0-beta.24066.3/build/Microsoft.DotNet.GenFacades.NotSupported.targets(56,5): error MSB4018: System.BadImageFormatException: Could not load file or assembly 'Microsoft.CodeAnalysis, Version=4.6.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. Reference assemblies cannot be loaded for execution. (0x80131058) [/__w/1/s/artifacts/sb/src/src/libraries/System.IO.Pipes.AccessControl/src/System.IO.Pipes.AccessControl.csproj::TargetFramework=net9.0]
  /__w/1/s/artifacts/sb/package-cache/microsoft.dotnet.genfacades/9.0.0-beta.24066.3/build/Microsoft.DotNet.GenFacades.NotSupported.targets(56,5): error MSB4018: File name: 'Microsoft.CodeAnalysis, Version=4.6.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' ---> System.BadImageFormatException: Could not load file or assembly 'Microsoft.CodeAnalysis, Version=4.6.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL'. Reference assemblies cannot be loaded for execution. (0x80131058) [/__w/1/s/artifacts/sb/src/src/libraries/System.IO.Pipes.AccessControl/src/System.IO.Pipes.AccessControl.csproj::TargetFramework=net9.0]
  /__w/1/s/artifacts/sb/package-cache/microsoft.dotnet.genfacades/9.0.0-beta.24066.3/build/Microsoft.DotNet.GenFacades.NotSupported.targets(56,5): error MSB4018: File name: 'Microsoft.CodeAnalysis, Version=4.6.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL' ---> System.BadImageFormatException: Cannot load a reference assembly for execution

**Comment by mthalman** [comment]
> Q: Will this fix the errors in this PR? [dotnet/runtime#97087](https://github.com/dotnet/runtime/pull/97087)

No, this change would only impact the behavior of the VMR build.

**Comment by ViktorHofer** [comment]
OK, I see. I just asked you for help directly in the PR: https://github.com/dotnet/runtime/pull/97087#issuecomment-1895965645

**Review by ViktorHofer — APPROVED** [review]

**Pull request merged**